src/daemon/jsonrpc/api.rs

```diff
@@ -219,26 +241,71 @@ impl RpcApi for RpcImpl {
     type Metadata = JsonRpcMetaData;
 
     fn stop(&self, meta: JsonRpcMetaData) -> jsonrpc_core::Result<()> {
+        meta.rpc_utils
+            .bitcoind_tx
+            .send(BitcoindMessageOut::Shutdown)
+            .map_err(|_| JsonRpcError::internal_error())?;
```

Comment:
Hmm. I acked using JSONRPC's internal error but we need the message. The `jsonrpc_core::Error` struct is public, therefore maybe we could have a helper creating it out of a string message?